models.cpp

```diff
@@ -17,6 +18,19 @@ double Model::atoms(unsigned int &size,
 
 
 
+Model::TermolecularNucleation::TermolecularNucleation()
+: TermolecularNucleation(std::numeric_limits<unsigned int>::signaling_NaN(),
+                         std::numeric_limits<unsigned int>::signaling_NaN(),
+                         std::numeric_limits<unsigned int>::signaling_NaN(),
+                         std::numeric_limits<unsigned int>::signaling_NaN(),
```

Comment:
There isn't a signaling NaN for `unsigned int`s: https://en.cppreference.com/w/cpp/types/numeric_limits/signaling_NaN

In deal.II, we define `unsigned int invalid_unsigned_int = static_cast<unsigned int>(-1)`, which when used as an index hopefully leads to an out-of-bounds error.

Reply:
Interesting. Your method makes sense. I'll make that change.